src/metabase/email.clj

```diff
@@ -95,12 +98,54 @@
                   (assert (#{:tls :ssl :none :starttls} (keyword new-value))))
                 (setting/set-value-of-type! :keyword :email-smtp-security new-value)))
 
+(defsetting email-rate-limit-per-second
+  (deferred-tru "The maximum number of emails that can be sent per second. The quotas are based on the total number of recipients")
+  :export?    true
+  :type       :integer
+  :visibility :settings-manager
+  :audit      :getter)
+
+(defn- make-email-throttler
+  [rate-limit]
+  (throttle/make-throttler
+   :email
+   :attempt-ttl-ms     1000
+   :initial-delay-ms   1000
+   :attempts-threshold rate-limit))
+
+(defonce ^:private email-throttler (when-let [rate-limit (email-rate-limit-per-second)]
+                                     (make-email-throttler rate-limit)))
+
+(defn check-email-throttle
+  "Check if the email throttler is enabled and if so, throttle the email sending based on the total number of recipients.
+
+  If the number of recipients exceeds the rate limit per second, we max-out the throttle limit but silience the exception.
+  This is to avoid creating regression bugs due to the throttling mechanism."
+  [email]
+  (when email-throttler
+    (let [recipients (set (mapcat #(% email) [:to :bcc]))
+          thorttle-check (fn []
```

Comment:
s/thorttle/throttle

Reply:
I'd also give this a bang as its sife-effecting and not parallel safe in tests.

Reply:
done!